test_suite/tests/derive.rs

```diff
@@ -50,7 +50,11 @@ fn struct_derive() {
 	let struct_type = Type::builder()
 		.path(Path::new("S", "derive"))
 		.type_params(tuple_meta_type!(bool, u8))
-		.composite(Fields::named().field_of::<bool>("t").field_of::<u8>("u"));
+		.composite(
+			Fields::named()
+				.field(Field::named_of::<bool>("t").with_type_display_name(vec!["T"]))
+				.field(Field::named_of::<u8>("u").with_type_display_name(vec!["U"])),
```

Comment:
It's a bit awkward to have to pass in a `Vec` to `with_type_display_name` but I guess it has to be a full path. Maybe you could allow `with_type_display_name("full::path::to::alias::T")` and split it up in `with_type_display_name()`? But that would likely mess up things elsewhere right?

Reply:
This is resolved now since the display name is no longer optional. The reason I made it this way was to optimize for composing the code in the macro (this code is ususally only written by hand in the tests), and to use a builder rather than creating many constructor permutations.